lib/puppet/functions/hash_values.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+# This function will return list of Hash values, the return value will be Array
+# NOTE : This function is expecting only Hash and return value will be Array
+#
+# @example :
+# $hash = {
+#   "key1" => "value1",
+#   "key2" => { "key2.1" => "value2.1"}
+# }
+# $hash.hash_value
+#
+# Output : ["value1", "value2.1"]
```

Comment:
:wave: As a Ruby developer, my feeling / experience is that the Puppet language quite match Ruby and my first reaction was that this new `hash_value()` function should be named `values` (to match `Hash#values`) and return `["value1", { "key2.1" => "value2.1"}]`.  Only then I could spot what this change was doing, and the function was precisely needing to work on the flattened hash values…

Naming things is hard, but IMHO we need to emphasis the special behavior of the function in its name, maybe adding something like `nested` or `recursive` in it?

Another possibility would be to add support for Hash to the `flatten` function, but this would break backwards compatibility :cold_face:.